python/ray/data/dataset_iterator.py

```diff
@@ -0,0 +1,167 @@
+import abc
+from typing import TYPE_CHECKING, Dict, List, Optional, Union, Iterator
+from typing_extensions import Literal
+
+from ray.data._internal.block_batching import BatchType
+from ray.data._internal.torch_iterable_dataset import TorchTensorBatchType
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    import tf
+    import torch
+
+
+@PublicAPI(stability="beta")
+class DatasetIterator(abc.ABC):
+    """An iterator for repeatedly reading a preprocessed dataset. Each
```

Comment:
```suggestion
    """An iterator for reading items from a Dataset or DatasetPipeline
    
    For Datasets, each iteration call represents a complete read of all items in the
    Datasets. For DatasetPipelines, each iteration call represents one pass (epoch)
    over the Dataset.
```

(I think this could be clarified--- here's a start of a suggestion).

Reply:
Ah right, I think this doc made a bit more sense when it was AIR only...